### ISSUE TITLE
Cannot use object of type Illuminate\Notifications\AnonymousNotifiable as array

### ISSUE THREAD

**Issue by michaelthedev** [itsgoingd/clockwork#724]
I have the code that sends a notification to a bunch of emails. The notification is queued also

```php
Notification::route('mail', $emailArray)
        ->notify($notification);
```

I have been getting this error
```
 "exception": "[object] (Error(code: 0): Cannot use object of type Illuminate\\Notifications\\AnonymousNotifiable as array at C:\\laragon\\htdocs\\github\\consultlink\\vendor\\itsgoingd\\clockwork\\Clockwork\\DataSource\\LaravelNotificationsDataSource.php:288)
```

I enabled queue logging and saw this.  The first one fails and the second one works well
![image](https://github.com/user-attachments/assets/97eaf0a6-32d7-4a27-891a-6f030d9c4e8d)

My stack: Laravel 11, PHP 8.3

**Comment by michaelthedev** [comment]
For context i am using On-Demand Notifications as explained here https://laravel.com/docs/11.x/notifications#on-demand-notifications

**Comment by michaelthedev** [comment]
Turns out it's partially an issue with Laravel
the on demand notifications still rely on the `via()` method inside it

I had this, and Laravel was attempting to send using both mail and broadcast even though i specified which route
```php
public function via(object $notifiable): array
    {
        return ['mail', 'broadcast'];
    }
```

**Comment by itsgoingd** [comment]
Fixed in Clockwork 5.3.2, thanks for the report!
